docs/CCNP Installation Guide.md

```diff
@@ -0,0 +1,157 @@
+
+## Build images
+
+Users can run the script, build\-CCNP\.sh, to quickly build images\. The script obtains the environment variables HTTP\_PROXY and HTTPS\_PROXY from the command line\. Therefore, please set up the proxy in advance or skip this if the proxy is not needed\. Please run the script on the master node which will download the latest CCNP code and build the images\.
+```
+./build-CCNP.sh
+```
+The script may take about 20 minutes, please be patient\. And it will generate four docker images: __ccnp\_eventlog\_server__, __ccnp\_measurement\_server__, __ccnp\-quote\-server__ __and__ __ccnp\-device\-plugin__\. You can use the command "docker images" to check it\.
+
+## Upload images
+
+In order for each node to pull these images, users need:
+
+1. Create a private image repository\. You can create an internal private image repository such as registry, harbor\.  And you can also use an external repository such as Docker Hub\.
+
+2. On the master node, add the private repository to the docker configuration file\. And on all nodes, add the private repository to the containerd configuration file\. Because users generate these images in docker on the master node and use these images in containerd on all nodes\.
+
+3. Tag these images \(ccnp\_eventlog\_server, ccnp\_measurement\_server, ccnp\-quote\-server and ccnp\-device\-plugin\) as private repository and upload them\.
+
+Take the example of creating an internal and password free registry repository\.
+```
+# Create a registry repo on the master node
+docker run --name=reqistry --restart=always -d -p 5000:5000 -v /opt/data/registry/:/var/lib/registry registry
+
+# Add the private repository to the docker configuration file on the master node
+# Add the following fields to the file /etc/docker/daemon.json
+"insecure-registries": ["<actual ip>:<port>"],
+
+# Make the configuration take effect
+sudo systemctl daemon-reload
+sudo systemctl restart docker
+
+# Add the private repository to the containerd configuration file on all nodes
+sudo mkdir -vp /etc/containerd/
+sudo touch /etc/containerd/config.toml
+sudo chmod +777  /etc/containerd/config.toml
+sudo containerd config default > /etc/containerd/config.toml
+# Add the following fields to the file /etc/containerd/config.toml
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."<actual ip>:<port>"]
+    endpoint = ["http://<actual ip>:<port>"]
+# Make the configuration take effect
+sudo systemctl daemon-reload
+sudo systemctl restart containerd.service
+
+# Tag these images as private repository and upload them
+docker tag ccnp_eventlog_server:0.1 <actual ip>:<port>/ccnp_eventlog_server:0.1
+docker push <actual ip>:<port>/ccnp_eventlog_server:0.1
```

Comment:
Since different private repos have different login methods, such as passwords, certificates, token, etc. So I take a passwordless repo as an example and this repository can be used directly without logging in.

Reply:
In CI, we use the "registry" docker image work with Kind as local-registry, it seems that it needs no password